Formula/podman.rb

```diff
@@ -34,10 +34,16 @@ class Podman < Formula
   depends_on "go-md2man" => :build
   depends_on "qemu"
 
+  # Remove hardcoded paths
+  patch :DATA
```

Comment:
Has this patch been upstreamed?

Reply:
Nope, upstream says it should be fixed by downstream...

i.e. they hardcoded some common brew paths

Reply:
The patch isn't strictly required, we can loop through all the unused paths first if required

Also the change is in a dependency (separate project), which is statically linked

Reply:
https://github.com/containers/common/blob/main/pkg/config/config_darwin.go

Reply:
Would the `inreplace` suffice without the patch? We don't carry patches that won't get accepted upstream. (With a handful of exceptions for patches that were made before this was made a policy, but even those could face the chopping block at some point.)

It also seems a bit weird to me for upstream to add Homebrew-specific paths that we would then remove here. Does a similar issue not affect other, non-Homebrew, users, who don't install into `/usr` or `/usr/local`?

Reply:
Most likely, unless they patch it.

Reply:
Then it seems to me that the right place to fix this is upstream, rather than for various downstream users to engage in duplicated efforts to 1) work out where the `gvproxy` path is hard-coded, 2) work out that they need to patch it for themselves, and 3) actually apply their patches whenever they update `podman`.

Reply:
See https://github.com/containers/podman/issues/12161#issuecomment-985645017

I removed the patches, so it will try the hardcoded locations before finding "libexec".

Also know that the upstream package uses libexec/podman (i.e. a subdirectory)

The right thing would be for "catatonit" and "gvproxy" to be dependencies, not bundled.

Reply:
(side note: catatonit is only used with podman, not with podman-remote)

Reply:
> See containers/podman#12161 (comment)

I did see that, but didn't we just establish above that this issue is not Homebrew-specific? Or was the conclusion above that this affects non-Homebrew users who install into non-standard locations incorrect?

> The right thing would be for "catatonit" and "gvproxy" to be dependencies, not bundled.

See https://github.com/Homebrew/homebrew-core/pull/87116#issuecomment-941081555. Though I'm not sure how having `gvproxy` be a dependency fixes the issue about hard-coded paths.

Reply:
I don't think they support Homebrew outside of standard prefix, or running on Linux (linuxbrew)
So all other Linux distributions have to patch the search paths in a similar way, or add symlinks.

Presumably if `gvproxy` was a system dependency, it would be able to find it in some path...
Currently it _is_ a separate package on Ubuntu - with a hardcoded path leading to Podman. :-(

Reply:
Actually it seems the standalone gvproxy path was already tried (and rejected), so "libexec" it is.
* https://github.com/Homebrew/homebrew-core/pull/83449

Reply:
> I don't think they support Homebrew outside of standard prefix, or running on Linux (linuxbrew)

Those are not separate projects anymore so they really just support half of Homebrew.

Reply:
It would be "fun" to see FreeBSDBrew, just like there was some stuff done for DarwinPorts

Basically we tried to run the basic system without the Apple bits, to keep it portable...
And without hardcoding "if apple else gnu". It was surprisingly difficult to do, though.